src/System.Management.Automation/engine/DataStoreAdapterProvider.cs

```diff
@@ -136,6 +143,7 @@ public string ModuleName
         internal void SetModule(PSModuleInfo module)
         {
             Module = module;
+            _fullName = null; // clear the cached FullName
```

Comment:
Please remove the comment. Better use `_cacheFullModuleName` name.

Reply:
But it isn't a full module name. That name would be misleading. But sure, I can remove the comment.

Reply:
If it is not a full name what about _cachedModuleName?

Reply:
The property is FullName, so it makes sense to me that the backing property is _fullName.

Reply:
I tried to insert your comment to the name that it is a cached :-)